### PR TITLE
B 19751 int

### PR DESCRIPTION
**Pull request by CoryKleinjanCACI**
## Summary

This task give the TOO role that ability to add shipments of any kind. The TOO will not have the ability to configure advance requests for PPMs.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the Office role
2. Login as a TOO
3. Select a move
4. navigate to the move details tab

### Repeat steps 5-6 for each shipment type (HHG, PPM, NTS, NTS-release)
5. Add a shipment
6. Verify the  shipment exists

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## Screenshots
Screenshot 2024-06-17 at 12 03 57 PM

### CODE REVIEW

**Comment by ajlusk** [comment]
As discussed with Cory, I believe the placement of the new button should be run past Danny. Changes are functioning as desired so far.

**Comment by antgmann** [comment]
Code changes look good and feature works nicely. Just waiting to hear on the button placement.

File: src/components/Office/ReviewSITExtensionModal/ReviewSITExtensionModal.jsx [hunk 1]
```diff
@@ -334,6 +334,7 @@ const ReviewSITExtensionsModal = ({ onClose, sitExtension, shipment, sitStatus,
                           <DropdownInput
                             label="Reason for edit"
                             name="requestReason"
+                            data-testid="reasonDropdown"
```
Comment (r-mettler) [inline src/components/Office/ReviewSITExtensionModal/ReviewSITExtensionModal.jsx:337]: Nice! The way we should be getting stuff for tests 😄

File: src/pages/Office/AddShipment/AddShipment.test.jsx [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+  jest.resetAllMocks();
+});
+
+describe('AddShipment component', () => {
+  describe('check different component states', () => {
+    it('renders the Loading Placeholder when the query is still loading', async () => {
+      useEditShipmentQueries.mockReturnValue(loadingReturnValue);
+      renderWithMocks();
+
+      const h2 = await screen.getByRole('heading', { name: 'Loading, please wait...', level: 2 });
```
Comment (r-mettler) [inline src/pages/Office/AddShipment/AddShipment.test.jsx:171]: Nit: await doesn't need to be here, at least not according to IDE.
image

File: src/pages/Office/AddShipment/AddShipment.test.jsx [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+      expect(h2).toBeInTheDocument();
+    });
+
+    it('renders the Something Went Wrong component when the query errors', async () => {
+      useEditShipmentQueries.mockReturnValue(errorReturnValue);
+
+      renderWithMocks();
+
+      const errorMessage = await screen.getByText(/Something went wrong./);
```
Comment (r-mettler) [inline src/pages/Office/AddShipment/AddShipment.test.jsx:180]: Same as above

**Review by r-mettler — APPROVED** [review]
Looks good.

**Pull request merged**